web/src/stream_create.ts

```diff
@@ -136,17 +142,29 @@ class StreamNameError {
         // realize the stream already exists, I may want to cancel.)
         const stream = stream_data.get_sub(stream_name);
         if (stream_name && stream) {
-            let error;
             if (stream.is_archived) {
-                error = $t({defaultMessage: "An archived channel with this name already exists."});
-                if (stream_settings_data.get_sub_for_settings(stream).can_change_name_description) {
-                    this.rename_archived_stream(stream.stream_id);
-                }
+                const can_rename =
+                    stream_settings_data.get_sub_for_settings(stream).can_change_name_description;
+                const can_view = stream_data.has_metadata_access(stream);
```

Comment:
Do we need to check metadata access here? I guess if client has the stream data in `stream_data.get_sub` then the user should have metadata access to the stream.